src/frontend/src/binder/insert.rs

```diff
@@ -113,10 +116,54 @@ impl Binder {
             }
         };
 
+        let mut target_table_col_idxs: Vec<ColumnIdx> = vec![];
+        for query_column in &columns {
+            let column_name = &query_column.value; // value or real_value() ?
+            let mut col_exists = false;
+            for (col_idx, table_column) in table_source.columns.iter().enumerate() {
+                if *column_name.to_lowercase() == table_column.name.to_lowercase() {
+                    target_table_col_idxs.push(ColumnIdx::from(col_idx));
+                    col_exists = true;
+                    break;
+                }
+            }
+            // Invalid column name found
+            if !col_exists {
+                return Err(RwError::from(ErrorCode::BindError(format!(
+                    "Column {} not found in table {}",
+                    column_name, table_source.name
+                ))));
+            }
+        }
+
+        // validate that query has a value for each target column, if target columns are used
+        // create table t1 (v1 int, v2 int);
+        // insert into t1 (v1, v2, v2) values (5, 6); // ...more target columns than values
+        // insert into t1 (v1) values (5, 6);         // ...less target columns than values
+        let (eq_len, msg) = match target_table_col_idxs.len().cmp(&expected_types.len()) {
+            std::cmp::Ordering::Equal => (true, ""),
+            std::cmp::Ordering::Greater => (false, "INSERT has more target columns than values"),
+            std::cmp::Ordering::Less => (false, "INSERT has less target columns than values"),
+        };
+        if !eq_len && !target_table_col_idxs.is_empty() {
+            return Err(RwError::from(ErrorCode::BindError(msg.to_string())));
+        }
+
+        // Check if column was used multiple times in query e.g.
+        // insert into t (v1, v1) values (1, 5);
+        let mut uniq_cols = target_table_col_idxs.clone();
+        uniq_cols.dedup();
```

Comment:
I'm afraid this method does not work as expected if the vec is not sorted. Do you want `all_unique`?

Reply:
For example, could you please test whether `insert into t (v1, v2, v1)` works?

Reply:
So I believe for the above example this should still work, because you then define more columns than there are in the table: 

```
insert into t1 (v1, v1) values (1, 5);
ERROR:  QueryError: Bind error: Column specified more than once
dev=>  insert into t1 (v1, v2, v1) values (1, 5);
ERROR:  QueryError: Bind error: INSERT has more target columns than values
```

However of cause you are right: It does not work as extended. Will change it. Thank you for the catch.